pull_request_template.md

```diff
@@ -0,0 +1,36 @@
+Thank you for working on PD! Please read PD's [CONTRIBUTING](https://github.com/pingcap/pd/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
+
+## What have you changed? (required)
+
+Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
+- Summarize your change (required)
+- How does this PR work? Need a brief introduction for the changed logic (optional)
+- Describe clearly one logical change and avoid lazy messages (optional)
+- Describe any limitations of the current code (optional)
+
+Please **NOTE** that:
+- Do not assume the code is self-evident
+- Do not assume reviewers understand the original issue
+
+## What are the type of the changes (required)?
+
+The currently defined types are listed below, please pick one of the types for this PR by removing the others:
+- New feature (non-breaking change which adds functionality)
+- Improvement (non-breaking change which is an improvement to an existing feature)
+- Bug fix (non-breaking change which fixes an issue)
+- Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## How has this PR been tested (required)?
```

Comment:
The word `required` is acceptable, but in other repos, the word `mandatory` is used. Change the several `required` to `mandatory` for consistency?

Reply:
I suppose `required` is more commonly used. I had looked up a dictionary to know what `mandatory` means...

Reply:
@disksing `required` is a synonym for `mandatory`. `mandatory` is generally used in the sense of "binding". Its usage is not limited to laws. For example, "mandatory training".

Reply:
IMO, _required_ is more commonly used to indicate whether a content must be filled in, contrast to _optional_. Here are some examples I can find:
- [HTML input required attribute](https://www.w3schools.com/TAgs/att_input_required.asp)
- [protocol buffers field rules](https://developers.google.com/protocol-buffers/docs/proto#specifying-field-rules)

I have also found a comparison between them on Quora: https://www.quora.com/Mandatory-compulsory-obligatory-essential-necessary-required-For-what-situations-are-each-if-these-words-best-suited 
Quoted below:
> Required suggests that a rule or procedure is operating. All visitors are required to wear a badge. Do you have to take the SAT? Yes, it's required if you're applying to college. 
> ...
> Mandatory means there is a mandate to perform a particular act in a particular situation. A flu shot is mandatory for all hospital staff. 

I suppose _required_ is more suitable to instruct that the questions must be answered in order to continue the pull request procedure.

Reply:
Thanks for your attention to the choice of words! You can keep it.